RichEditorView/Assets/editor/rich_editor.js

```diff
@@ -372,3 +373,67 @@ RE.getSelectedHref = function() {
 
     return href ? href : null;
 };
+
+/* Make sure all text nodes are wrapped in divs! */
+
+RE.wrapTextNodes = function() {
```

Comment:
without text nodes being wrapped in divs, it causes issues for calculating where the caret is.  I remember you telling me they were supposed to be divs, no?